packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -114,3 +123,27 @@ function writeServerCookie (event: H3Event, name: string, value: any, opts: Cook
     // else ignore if cookie doesn't exist in browser and value is null/undefined
   }
 }
+
+// custom ref that will update the value to undefined if the cookie expire
+function useCustomCookieRef<T> (value: T | undefined, delay?: number) {
+  let timeout: string | number | NodeJS.Timeout | undefined
+  return customRef((track, trigger) => {
+    return {
+      get () {
+        track()
+        return value
+      },
+      set (newValue) {
+        value = newValue
+        clearTimeout(timeout)
+        if (delay) {
+          timeout = setTimeout(() => {
+            value = undefined
+            trigger()
+          }, delay)
+        }
+        trigger()
+      }
+    }
+  })
+}
```

Comment:
I think there's a risk of having a memory leak here server-side. This should be probably client only i think?

Reply:
Also, I think on client side we might also want to clear the timeout when the component containing `useCookie` is unmounted...

Reply:
Thanks for your advices, I will work on it